ansible/roles/csm.rr.mgmt_nodes_placement_validation/files/mgmt_nodes_placement_validation.py

```diff
@@ -0,0 +1,257 @@
+
+    for rack_cnt, (rack_id, nodes) in enumerate(placements_dict.items()):
+        print("\nRack number: ", rack_cnt)
+        print("Rack ID: ", rack_id)
+        print("Management Nodes: ", nodes)
+
+        found=0
+        for node in nodes:
+            if re.match(r"^.*ncn-s00[0-9]$", node):
+                # Found storage node in a rack
+                storage_nodes_cnt+=1
+                found=1
+
+        if found == 0:
+            # Missing storage node in a rack
+            missing=1
+            missing_cnt+=1
+
+        rack_cnt+=1
+
+    print("\nTotal number of storage nodes: ", storage_nodes_cnt)
+
+    node_rack_missing_quorum=get_node_rack_missing_quorum(rack_cnt)
+
+    if storage_nodes_cnt >= min_storage_node_cnt:
+        if rack_cnt == min_rack_cnt and storage_nodes_cnt == min_master_node_cnt and  node_rack_missing_quorum >= node_rack_missing_quorum:
+            print(bcolors.FAIL + "\nOne or more racks missing storage node under minimum 3 rack condition" + bcolors.ENDC)
+            print(bcolors.FAIL + "\nPlease re arrange the storage nodes for equal distribution" + bcolors.ENDC)
+            print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            sys.exit(1)
+        elif rack_cnt > min_rack_cnt and storage_nodes_cnt >= min_master_node_cnt and node_rack_missing_quorum > node_rack_missing_quorum:
+            print(bcolors.FAIL + "\nMore than one rack missing storage node in a", rack_cnt , "rack system" + bcolors.ENDC)
+            print(bcolors.FAIL + "\nPlease re arrange the storage nodes for equal distribution" + bcolors.ENDC)
+            print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+            sys.exit(1)
+     else:
+        print(bcolors.FAIL + "\nRack Resiliency can not be enabled with", storage_nodes_cnt, "storage nodes" + bcolors.ENDC)
+        print(bcolors.FAIL + "\nNeed minimum of 3 storage nodes configured" + bcolors.ENDC)
+        print(bcolors.FAIL + "Exiting placement validation...\n"  + bcolors.ENDC)
+        sys.exit(1)
+
+def validate_compute_uan_nodes_placement(placements_dict):
+    rack_cnt=1
+    managed_nodes_cnt=0
+    missing_cnt=0
+
+    print(bcolors.OKGREEN + bcolors.UNDERLINE + "\nValidate Managed nodes (Compute and UANs)"+ bcolors.ENDC)
+
+    for rack_cnt, (rack_id, nodes) in enumerate(placements_dict.items()):
+        print("\nRack number: ", rack_cnt)
+        print("Rack ID: ", rack_id)
+        print("Management/ Managed Nodes: ", nodes)
+
+        found=0
+        for node in nodes:
+            if re.match(r"^.*nid00000[0-9]$", node) or re.match(r"^.*uan0[0-9]$", node):
+                print(bcolors.WARNING + "Found", node, "Compute/ UAN node under management rack" + bcolors.ENDC)
+                managed_nodes_cnt+=1
+                found=1
+
+        if found == 1:
+            print(bcolors.WARNING +  "\nOne or more Managed nodes found under management Racks" + bcolors.ENDC)
```

Comment:
This is inside the main for loop, which goes over each rack. So it could potentially be printed for each rack. In that case, it should include something that identifies the specific rack, or else it should be moved outside of the for loop so it only gets printed once. Yes, we do print the rack number before each loop, but people may just be grepping for warnings or errors -- so those should contain all of the relevant info, ideally.